src/briefcase/integrations/__init__.py

```diff
@@ -13,7 +13,93 @@
     xcode,
 )
 
+
+class ToolCache:
+    def __init__(self):
+        # Tool wrappers with interfaces
+        self.android_sdk = None
+        self.docker = None
+        self.flatpak = None
+        self.git = None
+        self.jdk = None
+        self.linuxdeploy = None
+        self.rcedit = None
+        self.visualstudio = None
+        self.wix = None
+
+        # Flags for whether a tool is verified
+        self._xcode = False
+        self._xcode_cli = False
+
+        # App specific tools
+        self._current_app = None
```

Comment:
Yes; although, I think, in practice, that's all that is technically necessary (right now) since I believe all of the commands for an app are processed before moving to the next app.

Whether that is or isn't the case, though, this was definitely a weakness of my original implementation.

Reply:
I think you're technically correct (at least at the moment) - but it seems like a potential footgun. It wouldn't take much of a change to alter the ordering of command processing, at which point the caching is completely invalidated.

Reply:
Obsolete with `ToolCache` refactoring. App-bound tools are now managed independently of each other.